src/main/java/com/android/volley/Request.java

```diff
@@ -124,6 +125,8 @@
     @GuardedBy("mLock")
     private NetworkRequestCompleteListener mRequestCompleteListener;
 
+    private Map<String, Object> mTags = new HashMap<>();
```

Comment:
While this might be reasonable given that there's already an mTag, I think it would be unclear to clients how these interact with each other given the same naming.

I'm also not as convinced this piece is as general purpose. The existing tag serves the purpose of making it possible to cancel all requests with a given tag. This does nothing except store app-specific metadata, and in cases like this I'd generally be asking client apps to manage this in their own request subclasses instead of adding it to Volley itself. What does this end up being used for, and is there another way to keep track of this externally instead of in the request itself?

Reply:
The idea behind the tags here is similar to View tags (https://developer.android.com/reference/android/view/View.html#setTag(java.lang.Object), https://developer.android.com/reference/android/view/View.html#setTag(int,%20java.lang.Object)) - I would like to store additional data associated with the request along with the request.

Although it's still possible to do this externally (e.g., with WeakHashMap), having it here greatly simplifies the design.

And unfortunately storing this in a subclass won't help, as in our specific case we want to have this available for all the requests, no matter whether they are extending our subclass or not.

Having said that I don't feel too strongly, because having the listener APIs will still simplify things. I can revert this part for now and we'll see how it works out without it.

Reply:
Removed this for now